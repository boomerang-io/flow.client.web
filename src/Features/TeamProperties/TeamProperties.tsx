import React, { useState } from "react";
import { Helmet } from "react-helmet";
import { useHistory } from "react-router-dom";
import { useQuery } from "react-query";
import { useAppContext } from "Hooks";
import TeamPropertiesTable from "./TeamPropertiesTable";
import { serviceUrl, resolver } from "Config/servicesConfig";
import { appLink } from "Config/appConfig";
import styles from "./teamProperties.module.scss";

function TeamProperties() {
  const history = useHistory();
  const { activeTeam } = useAppContext();

  /** Get team properties */
  const teamPropertiesUrl = serviceUrl.getTeamProperties({ id: activeTeam?.id });
  const {
    data: propertiesData,
    isLoading,
    error: propertiesError,
  } = useQuery(teamPropertiesUrl, resolver.query(teamPropertiesUrl), { enabled: Boolean(activeTeam?.id) });

  /** Check if there is an active team or redirec to home */
  if (!activeTeam) {
    return history.push(appLink.home());
  }

  return (
    <div className={styles.container}>
      <Helmet>
        <title>Team Parameters</title>
      </Helmet>
      <TeamPropertiesTable
        properties={propertiesData ?? []}
        propertiesAreLoading={isLoading}
        propertiesError={propertiesError}
        //@ts-ignore
        activeTeam={activeTeam}
      />
    </div>
  );
}

export default TeamProperties;
