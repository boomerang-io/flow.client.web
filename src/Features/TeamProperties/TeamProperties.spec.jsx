import React from "react";
import TeamProperties from "./index";
import { fireEvent, screen, waitFor } from "@testing-library/react";
import { startApiServer } from "ApiServer";

let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
});

describe("TeamProperties --- Snapshot Test", () => {
  test("Capturing Snapshot of TeamProperties", () => {
    const { baseElement } = rtlContextRouterRender(<TeamProperties />);
    expect(baseElement).toMatchSnapshot();
  });
});

describe("TeamProperties --- RTL", () => {
  test("Selects team from dropdown and shows table data", async () => {
    rtlContextRouterRender(<TeamProperties />);
    const dropDown = await screen.findByPlaceholderText("Select a team");
    fireEvent.click(dropDown);
    await waitFor(() => {
      screen.getByText("IBM Services Engineering");
    });

    fireEvent.click(screen.getByText("IBM Services Engineering"));
    const rowItem = await screen.findByText("for testing purpose");
    expect(rowItem).toBeInTheDocument();
  });
});
