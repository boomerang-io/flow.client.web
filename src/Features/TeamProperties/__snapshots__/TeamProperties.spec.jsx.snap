// Vitest Snapshot v1

exports[`TeamProperties --- Snapshot Test > Capturing Snapshot of TeamProperties 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="_container_d032fc"
    >
      <header
        class="cds--bmrg-feature-header _header_d292d6"
      >
        <section
          class="cds--bmrg-feature-header__content"
        >
          <div
            class="cds--bmrg-feature-header__header"
          >
            <h1
              class="cds--bmrg-feature-header-title _headerTitle_d292d6"
            >
              Team Parameters
            </h1>
            <p
              class="cds--bmrg-feature-header-title"
            >
              Set team parameters that are accessible in all workflows for that team.
            </p>
          </div>
        </section>
      </header>
      <div
        class="_tableContainer_d292d6"
      >
        <div
          class="_tableHeader_d292d6"
        >
          <div
            class="_dropdown_d292d6"
          >
            <div
              class="cds--bmrg-select"
            >
              <div
                class="cds--list-box__wrapper"
              >
                <label
                  class="cds--label"
                  dir="auto"
                  for="team-parameters-select"
                  id="downshift-0-label"
                >
                  <div
                    style="display: flex;"
                  >
                    <div>
                      Teams
                    </div>
                  </div>
                </label>
                <div
                  class="cds--combo-box cds--list-box"
                >
                  <div
                    class="cds--list-box__field"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      autocomplete="off"
                      class="cds--text-input cds--text-input--empty"
                      data-testid="team-parameters-combobox"
                      id="team-parameters-select"
                      placeholder="Select a team"
                      role="combobox"
                      tabindex="0"
                      title=""
                      type="text"
                      value=""
                    />
                    
                    <button
                      aria-haspopup="true"
                      aria-label="Open"
                      class="cds--list-box__menu-icon"
                      data-toggle="true"
                      role="button"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    aria-label="Choose an item"
                    class="cds--list-box__menu"
                    id="downshift-0-menu"
                    role="listbox"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1e9d5ym"
        >
          <div
            class="css-vurnku"
          >
            <div
              class="cds--bmrg-error-page"
            >
              <h1
                class="cds--bmrg-error-page__title"
              >
                Select a team
              </h1>
              <section
                class="cds--bmrg-error-page__graphic"
              >
                <svg
                  viewBox="0 0 152.94 211.02"
                >
                  <defs>
                    <style>
                      .prefix__cls-1{fill:#3ddbd9}.prefix__cls-2{fill:#c1c7cd}.prefix__cls-3{fill:#009d9a}.prefix__cls-4{fill:#121619}
                    </style>
                  </defs>
                  <title>
                    Asset 9
                  </title>
                  <g
                    data-name="Layer 2"
                    id="prefix__Layer_2"
                  >
                    <g
                      data-name="Layer 1"
                      id="prefix__Layer_1-2"
                    >
                      <path
                        class="prefix__cls-1"
                        d="M111.9 40.73c-.06.19-.12.37-.19.55l-.16.32-.18.29c-.19.23-.39.44-.59.65a12.07 12.07 0 01-2.69 2c-6.26 3.74-14 5.17-21.15 6.18a120 120 0 01-27.83.69c-2-.19-4-.45-6-.77a11.19 11.19 0 00-7.28-3.81 20.25 20.25 0 00-4.25.06c-.08-.82-.17-1.65-.25-2.48A119.64 119.64 0 0061 47.6a107 107 0 0021.91 0 70 70 0 0019.61-5.43 60.34 60.34 0 007.83-4c.65-.4 1.3-.83 1.94-1.27a13.16 13.16 0 01-.39 3.83zM117.22 116.86a81 81 0 018 10.23c.21.31.41.63.61.95a34.31 34.31 0 008.75-10.1 36.87 36.87 0 004.42-10.11 18.36 18.36 0 01-6.31 2.27c-1 .14-1.88-1-1.88-1.87v-1.06a38.75 38.75 0 01-13.59 9.69z"
                      />
                      <path
                        class="prefix__cls-1"
                        d="M137.05 158.7c-3.19-6.4-5.78-13.09-9.05-19.46a86.76 86.76 0 00-14.61-21 30.5 30.5 0 01-7 .75 78.83 78.83 0 01-10.86-.61 75.79 75.79 0 01-11.08-2 23.18 23.18 0 01-7.8-3.28 1.92 1.92 0 01-.67-2.57 1.88 1.88 0 012.56-.67 17.47 17.47 0 005.69 2.58c.2-2.34 1.64-4.57 3-6.34 4.86-6.16 13.11-9.18 20.82-8.66a24 24 0 0112.49 4.68 20.23 20.23 0 015.11 5 30.81 30.81 0 005.8-7c5.64-9.54 6-22.08 1.48-32.14-14.36 6.65-30.7 8.84-46.37 9.48a250 250 0 01-33.62-1.06c-7.12-.67-14.26-1.55-21.3-2.92a34.88 34.88 0 01-7.26 7.69c-1.58 1.17-3.27 2.2-5 3.23 2.88.2 5.78.11 8.7.32a1.9 1.9 0 01.95 3.5 16.66 16.66 0 00-5.7 7.78l.36-.26c1.52-1 3.42-2.59 5.29-2.89A1.89 1.89 0 0130.8 96c.36-.44-.06.37 0 0 0 .19-.11.37-.16.55a8.21 8.21 0 00-.23 1.31 32.68 32.68 0 00-.13 3.57c0 1 .06 2 .12 3 1.7.51 3.4 1.11 5.08 1.69.55-2.61 1.11-5.23 1.66-7.84a1.9 1.9 0 012.76-1.13 40.62 40.62 0 016.54 6A76.27 76.27 0 0159 121.93c1.05 2.16-2.19 4.06-3.23 1.89a73.22 73.22 0 00-12-18.09 53.584 53.584 0 00-3.65-3.58q-.72 3.45-1.46 6.89a1.87 1.87 0 01-2.3 1.31l-4.93-1.59-2.57-.76c.1.43.19.87.29 1.3a17.49 17.49 0 01.77 4.69c-.15 2.35-2.66 2.16-4.39 2.2h-5.87C25.5 127.55 37.89 134 50.15 136.3c2.38.44 1.37 4.06-1 3.61a54.4 54.4 0 01-17.52-6.47 114.82 114.82 0 00.37 13.63A37.75 37.75 0 0037.13 163a36.11 36.11 0 002.93 4 3.92 3.92 0 011 2.47c.17 1.34.25 2.68.35 4 .51 6.23 1.46 14.11 6.43 18.39.14-2.27.44-4.52.74-6.64a1.89 1.89 0 012.29-1.22 2 2 0 011.31 2.31 45.16 45.16 0 00-.49 10.45 16.31 16.31 0 00.5 2.86c.12.42.25.84.4 1.25a5.08 5.08 0 00.32.71 7 7 0 002.17 2.67l.08.07.31.23.68.46.59.37v-5.27a1.89 1.89 0 012.83-1.61l7.65 5.09c0-1.33-.09-2.65-.14-4a1.89 1.89 0 012.38-1.81l7.55 3a20.56 20.56 0 00-.51-2.5 13.36 13.36 0 00-6-8.57 1.91 1.91 0 01-.67-2.57 21.58 21.58 0 012.64-3.39 47.91 47.91 0 0117.35-11.65c5.78-2.39 12.29-3.43 17.68-6.68a1.91 1.91 0 012.76 1.12 13 13 0 006 6.75 21.7 21.7 0 01-1.34-6.36 1.84 1.84 0 012.37-1.81c3 1 6.08 2 9.16 2.89.41-.67.81-1.34 1.2-2 1.12-1.91 2.17-3.9 3.41-5.73.66-1 1.38-1.69 2.66-1.38a6.77 6.77 0 013.43 2.38 21.27 21.27 0 012.4 3.55 1.74 1.74 0 00.1.19l.76-.62a35.55 35.55 0 01-3.36-5.7z"
                      />
                      <path
                        class="prefix__cls-2"
                        d="M40.89 40.47c1.91.51 3.86.93 5.8 1.32A129.54 129.54 0 0062.62 44a97.38 97.38 0 0021.52-.34 67 67 0 0018.14-5.5c3.26-1.52 6.92-3.28 9.54-5.7-.08-.51-.16-1-.26-1.54a2 2 0 01-.08-.4 51.89 51.89 0 00-6.3-16.59c-1.34-2.22-3.06-4.59-5.13-5.79a29.86 29.86 0 00-6.94-2.58 48.58 48.58 0 00-7.57-1.43A60.74 60.74 0 0074 4c-.55 0-1.11.1-1.66.16h-.39c-.24 0-.49.08-.73.13a18 18 0 00-4.51 1.58 14.22 14.22 0 00-4.12 3 6.87 6.87 0 00-1.33 2c-.33.78-.23 1.67-.55 2.42-1.18 2.81-6.06 2.16-8.37 1.85A51.68 51.68 0 0143.17 13c-.4-.14-.8-.28-1.2-.44a5.11 5.11 0 00-.4.43 7.75 7.75 0 00-.57 1.1 15.37 15.37 0 00-.82 2.57c-1.52 6.19-.4 12.68.23 18.92z"
                      />
                      <path
                        class="prefix__cls-2"
                        d="M149.06 43.53a11.89 11.89 0 00-4.41-6.37 36.91 36.91 0 00-7.77-4.62c-3.47-1.63-7.23-3.28-11.09-3.64a12.13 12.13 0 00-3.65.15 15.12 15.12 0 00-1.62.49c-.31.12-.59.24-1.14.51a31.44 31.44 0 01-3.89 1.7l.12.72a1.79 1.79 0 01.12.83c.46 3.16.63 6.58-.6 9.46-1.43 3.36-5.36 5.4-8.54 6.78A68.21 68.21 0 0190.73 54a120 120 0 01-34.48.95 30.52 30.52 0 013.57 7.92c.1 0 .2 0 .3-.06 2.36-.51 3.36 3.11 1 3.62A40 40 0 0142.23 66a29.62 29.62 0 01-7.7-3.11 1.93 1.93 0 01.82-3.52c-.45-.75-.9-1.5-1.32-2.27a13.14 13.14 0 01-1.64-3.88c-.3-1.68.65-2.93 2-3.83a13.89 13.89 0 013.51-1.61l-.36-3.5c-.65.34-1.28.72-1.91 1.11a65.6 65.6 0 01-11.93 6.33 50.25 50.25 0 01-19.95 2.8 7.85 7.85 0 000 .84v.35c0 .22.09.44.15.65s.13.43.21.65a.44.44 0 010 .08l.09.17a7.42 7.42 0 001.5 2.11 20.24 20.24 0 005.83 4c6.47 3.34 13.74 5.01 20.77 6.37a233.69 233.69 0 0029.83 3.57 223 223 0 0030.91 0 139.68 139.68 0 0024-3.77 76.5 76.5 0 0017-6.28 33.2 33.2 0 0012.63-11c1.64-2.52 3.06-5.7 2.39-8.73zm-18 8.3a29.68 29.68 0 01-1 3.8 52.44 52.44 0 00-7.5-5.71c-.8-.52-1.61-1-2.43-1.48a18.16 18.16 0 01.94-4.82 4 4 0 012.8-2.56 20.56 20.56 0 014.86 2.94 6.94 6.94 0 012.35 3.43c0 .2.1.4.14.6 0 0 0 .19.05.28a9.87 9.87 0 01.06 1.22 10.21 10.21 0 01-.25 2.3z"
                      />
                      <path
                        class="prefix__cls-3"
                        d="M51.1 54.06a1.44 1.44 0 01-.68-.79 10.39 10.39 0 00-1.86-1.66 7 7 0 00-4.13-1 18.62 18.62 0 00-5.14.74 14.89 14.89 0 00-3.15 1.43 5.23 5.23 0 00.37.95c.62 1.29 1.34 2.52 2.08 3.74q1.36 2.24 2.87 4.39c.48.15 1 .3 1.49.44a35.45 35.45 0 0013.12 1.1A29.58 29.58 0 0051.8 55c-.22-.33-.46-.64-.7-.94zM143.89 166.33c-1.46 1.22-4.83 5.1-6.57 2.32-1-1.58-1.95-4.4-3.58-5.63l-.12.19c-.56.93-1.11 1.88-1.66 2.83-.93 1.6-1.8 3.28-2.87 4.79-1.29 1.81-3.21.93-5 .4s-3.34-1-5-1.58c0 .2.08.4.14.6a16.57 16.57 0 002.59 5.73 1.13 1.13 0 01.15.21c.18.23.36.45.56.67 4.12 4.63 11.93 5.08 17.14 1.71a14.5 14.5 0 005.18-6.4 23.76 23.76 0 001.2-3.63 19.34 19.34 0 00.54-3.24c0-.43 0-.86.06-1.29zM121.35 107.78a20.61 20.61 0 00-4.35-3.55 21.27 21.27 0 00-21.53-.31 18.34 18.34 0 00-5.79 5.1 14 14 0 00-.88 1.38c-.11.2-.22.4-.32.61 0 .06-.13.28-.18.37a8.35 8.35 0 00-.35 1.07 2.3 2.3 0 00-.06.28v.14a3 3 0 01-.06.46 78.22 78.22 0 0010.32 1.6c6 .55 12.15.89 17.72-1.54a36 36 0 006.79-3.84 9.5 9.5 0 00-1.31-1.77zM130 55.63a29.68 29.68 0 001-3.8 10.21 10.21 0 00.25-2.35 9.87 9.87 0 00-.06-1.22c0-.09 0-.24-.05-.28 0-.2-.09-.4-.14-.6a6.94 6.94 0 00-2.27-3.38 20.56 20.56 0 00-4.88-2.89 4 4 0 00-2.8 2.56 18.16 18.16 0 00-.94 4.82c.82.47 1.63 1 2.43 1.48a52.44 52.44 0 017.46 5.66z"
                      />
                      <path
                        d="M77.82 81.62a1.91 1.91 0 00-1.12-.87h-.26a1.8 1.8 0 00-1-.31 1.88 1.88 0 00-1.33.55 1.9 1.9 0 00-.55 1.32v3.16a1.9 1.9 0 00.55 1.33l.38.29a1.86 1.86 0 001 .26h.21a3.19 3.19 0 00.69-.1 3.34 3.34 0 001.14-.66 3.22 3.22 0 001-1.49 3.69 3.69 0 000-2 6.23 6.23 0 00-.71-1.48zM125.25 78.9a2.82 2.82 0 00-.51-.68 2 2 0 00-1.33-.55 1.32 1.32 0 00-.34 0 1.84 1.84 0 00-2.46.15l-.29.38a1.86 1.86 0 00-.26 1 17.92 17.92 0 00-.09 2.07v.94a3.2 3.2 0 00.32 1.21 2.3 2.3 0 00.85.9 2.52 2.52 0 002 .26 3 3 0 001.61-1.25 5.45 5.45 0 00.53-1.15 4.83 4.83 0 00.22-.74 4.58 4.58 0 000-1.73 2.89 2.89 0 00-.25-.81zM128 38.94c-.72-.41-1.46-.77-2.21-1.12a3.74 3.74 0 00-3-.36 7.91 7.91 0 00-3.56 2.16c-1.73 1.92-2.32 4.48-2.64 7a3.36 3.36 0 00-1-.2c-2.31 0-2.41 3.38-.31 3.71h.27-.24.08a1 1 0 00.26.07 9.91 9.91 0 011.17.61c1.38.72 2.71 1.53 4 2.38a41 41 0 018.63 7.13 1.63 1.63 0 00.81.47h.26a1.65 1.65 0 00.94-.12 1.73 1.73 0 001.09-1.16v-.07c1.83-4.75 3.93-10.65 1.15-15.46a14.69 14.69 0 00-5.7-5.04zm-5.43 11c-.8-.52-1.61-1-2.43-1.48a18.16 18.16 0 01.94-4.82 4 4 0 012.8-2.56 20.56 20.56 0 014.85 2.92 6.94 6.94 0 012.35 3.43c0 .2.1.4.14.6 0 0 0 .19.05.28a9.87 9.87 0 01.06 1.22 10.21 10.21 0 01-.25 2.35 29.68 29.68 0 01-1 3.8 52.44 52.44 0 00-7.54-5.76z"
                      />
                      <path
                        d="M146.28 58.64c2.92-3.15 5.79-7 6.49-11.33a12.25 12.25 0 00-2.05-9.13c-4-5.62-11-8.81-17.29-11.15-3.64-1.37-7.58-2.37-11.48-1.74a23.74 23.74 0 00-6.38 2.48l-.38.17-.43.16a62.56 62.56 0 00-4-11.53c-1.8-3.93-4.1-8.08-7.57-10.78-2.37-1.88-5.54-2.79-8.43-3.67A53.79 53.79 0 0084 .23 51.27 51.27 0 0070.45.7c-4.62.89-10 3.65-12.26 8a9.34 9.34 0 00-.92 2.79 3.83 3.83 0 01-.73.14 15.82 15.82 0 01-3-.07 42.38 42.38 0 01-6.69-1.34c-.88-.23-1.74-.48-2.6-.76l-.87-.3h-.09c-.07 0-.52-.23-.61-.26-2.36-.67-4.08 1.43-5 3.34-2.2 4.79-2.36 10.35-1.87 15.51.37 4 .79 8 1.2 12.07l.06.51c-2.53.81-4.88 2.82-7 4.14a43 43 0 01-14.56 5.63 53.54 53.54 0 01-13.26.6 1.91 1.91 0 00-1.81 1.38c-1.51 4.79 1.07 9 4.8 11.86 5.31 4.06 12.16 6.13 18.55 7.75 1.27.32 2.54.62 3.82.91a27.1 27.1 0 01-5.5 5.56c-2.88 2-6 3.71-9 5.66a1.9 1.9 0 00.45 3.43 46.18 46.18 0 0010.05 1 17.93 17.93 0 00-4.72 11.88c0 1.29 1.63 2.51 2.82 1.62s2.4-1.74 3.62-2.57c.42-.28.84-.57 1.27-.84-.05.78-.05 1.56-.06 2.29q0 1.45.06 2.91a1.92 1.92 0 00-2 2 11.82 11.82 0 00.46 2.71c.27 1.21.56 2.42.81 3.63 0 .18.07.36.11.54h-9.24a1.88 1.88 0 00-1.81 2.37 37.64 37.64 0 0013 16.25c0 5-.25 9.91.17 14.85a44.65 44.65 0 004.61 17 34.54 34.54 0 002.33 3.8c.49.7 1 1.39 1.55 2.06.11.14.34.5.55.75 0-.11.28-.14.17.18l.11.07-.11-.06V169.98l.09 1c.07.77.13 1.54.19 2.32a53.8 53.8 0 002.07 12.35c1.44 4.45 4 8.63 8.16 10.93a19.09 19.09 0 001.23 5.61 13.24 13.24 0 005.7 6.44c.51.33 1 .66 1.52 1l.08.06a3.36 3.36 0 01.29.24 6.25 6.25 0 01.62.64c1.09 1.27 3.2.21 3.2-1.33v-5.57l7.77 5.18a1.89 1.89 0 002.82-1.62l-.15-4.76 7.6 3a1.9 1.9 0 002.38-1.81c.21-5.9-2.15-12.28-6.84-16A26.23 26.23 0 0176 185.5a43 43 0 0110.54-7.75 75.13 75.13 0 0112.57-4.95 59.13 59.13 0 008.45-3.14c2.36 4.48 7 7.68 11.83 9.19a16.27 16.27 0 0014.36 5.51c7-.67 12.21-5 14.83-11.36a26.36 26.36 0 001.79-7.53 11.9 11.9 0 00-.31-4.9 2.26 2.26 0 00-2.72-1.66c-1.1.32-2.1 1.45-3 2.16l-1.08.9a34 34 0 01-2.71-4.59c-1.45-2.86-2.73-5.8-4-8.72a169.52 169.52 0 00-8.7-17.43 38.27 38.27 0 0010-11.4c2.84-4.73 5.1-10.09 5.47-15.65a1.89 1.89 0 00-2.82-1.62c-1 .63-2 1.21-3 1.77-.85.46-1.88 1-2.88 1.44v-3.71A37.59 37.59 0 00139.7 85a40 40 0 00-3.44-18.71 39.17 39.17 0 0010.02-7.65zM111.9 40.73c-.06.19-.12.37-.19.55l-.16.32-.18.29c-.19.23-.39.44-.59.65a12.07 12.07 0 01-2.69 2c-6.26 3.74-14 5.17-21.15 6.18a120 120 0 01-27.83.69c-2-.19-4-.45-6-.77a11.19 11.19 0 00-7.28-3.81 20.25 20.25 0 00-4.25.06c-.08-.82-.17-1.65-.25-2.48A119.64 119.64 0 0061 47.6a107 107 0 0021.91 0 70 70 0 0019.61-5.43 60.34 60.34 0 007.83-4c.65-.4 1.3-.83 1.94-1.27a13.16 13.16 0 01-.39 3.83zM36 52.82c0 .08 0 .14 0 0s-.22.18 0 0zm5.46 9q-1.5-2.16-2.87-4.39c-.74-1.22-1.46-2.45-2.08-3.74a5.23 5.23 0 01-.37-.95 14.89 14.89 0 013.15-1.43 18.62 18.62 0 015.14-.74 7 7 0 014.13 1 10.39 10.39 0 011.86 1.66 1.44 1.44 0 00.68.79c.24.3.48.61.7.92a29.58 29.58 0 014.27 8.42A35.45 35.45 0 0143 62.26c-.57-.14-1.06-.26-1.54-.44zM42 12.44zm-.41 0l-.16-.08zm-1.48 4.2a15.37 15.37 0 01.89-2.55 7.75 7.75 0 01.53-1.09 5.11 5.11 0 01.4-.43c.4.16.8.3 1.2.44a51.68 51.68 0 009.19 2.12c2.31.31 7.19 1 8.37-1.85.32-.75.22-1.64.55-2.42a6.87 6.87 0 011.33-2 14.22 14.22 0 014.12-3 18 18 0 014.51-1.58c.24-.05.49-.1.73-.13h.39c.55-.06 1.11-.12 1.66-.16a60.74 60.74 0 0111.52.09 48.58 48.58 0 017.57 1.43 29.86 29.86 0 016.94 2.58c2.07 1.2 3.79 3.57 5.13 5.79a51.89 51.89 0 016.3 16.59 2 2 0 00.08.4c.1.51.18 1 .26 1.54-2.62 2.42-6.28 4.18-9.54 5.7a67 67 0 01-18.14 5.5 97.38 97.38 0 01-21.48.39 129.54 129.54 0 01-15.93-2.2c-1.94-.39-3.89-.81-5.8-1.32l-.51-4.89c-.63-6.25-1.75-12.74-.23-18.93zm-36.4 39s0-.07 0 0zm23.6 56.71s0-.08.11 0 .06 0-.06 0zm-1.06-5.07c-.15 0-.12 0 0 0zM56.4 209.59zm89.67-41.05a23.76 23.76 0 01-1.2 3.63 14.5 14.5 0 01-5.18 6.4c-5.21 3.37-13 2.92-17.14-1.71-.2-.22-.38-.44-.56-.67a1.13 1.13 0 00-.15-.21 16.57 16.57 0 01-2.59-5.73c-.06-.2-.1-.4-.14-.6q2.49.81 5 1.58c1.77.53 3.69 1.41 5-.4 1.07-1.51 1.94-3.19 2.87-4.79.55-.95 1.1-1.9 1.66-2.83l.12-.19c1.63 1.23 2.59 4 3.58 5.63 1.74 2.78 5.11-1.1 6.57-2.32l2.78-2.32c0 .43 0 .86-.06 1.29a19.34 19.34 0 01-.56 3.24zm-18.82-.32c.17 0 .41.13 0 0zm11.21-2.22c-.33.31-.54.08 0 0zm8.06-4.1c-.2-.36-.05-.25 0-.05zm-7.5-54a36.87 36.87 0 01-4.39 10.11 34.31 34.31 0 01-8.75 10.1c-.2-.32-.4-.64-.61-.95a81 81 0 00-8-10.23 38.75 38.75 0 0013.62-9.69v1.06c0 .91.87 2 1.88 1.87a18.36 18.36 0 006.23-2.34zm-51.13 5v-.14a2.3 2.3 0 01.06-.28 8.35 8.35 0 01.35-1.07c0-.09.15-.31.18-.37.1-.21.21-.41.32-.61a14 14 0 01.88-1.38 18.34 18.34 0 015.79-5.1 21.27 21.27 0 0121.53.31 20.61 20.61 0 014.35 3.55 9.5 9.5 0 011.31 1.77 36 36 0 01-6.79 3.84c-5.57 2.43-11.67 2.09-17.72 1.54a78.22 78.22 0 01-10.32-1.6 3 3 0 00.06-.49zm43.56-12.8a30.81 30.81 0 01-5.8 7 20.23 20.23 0 00-5.11-5 24 24 0 00-12.49-4.68c-7.71-.52-16 2.5-20.82 8.66-1.41 1.77-2.85 4-3 6.34a17.47 17.47 0 01-5.69-2.58 1.88 1.88 0 00-2.56.67 1.92 1.92 0 00.67 2.57 23.18 23.18 0 007.8 3.28 75.79 75.79 0 0011.08 2 78.83 78.83 0 0010.86.61 30.5 30.5 0 007-.75 86.76 86.76 0 0114.61 21c3.27 6.37 5.86 13.06 9.05 19.46a35.55 35.55 0 003.36 5.67l-.76.62a1.74 1.74 0 01-.1-.19 21.27 21.27 0 00-2.4-3.55 6.77 6.77 0 00-3.43-2.38c-1.28-.31-2 .42-2.66 1.38-1.24 1.83-2.29 3.82-3.41 5.73-.39.68-.79 1.35-1.2 2-3.08-.87-6.12-1.9-9.16-2.89a1.84 1.84 0 00-2.37 1.81 21.7 21.7 0 001.34 6.36 13 13 0 01-6-6.75 1.91 1.91 0 00-2.76-1.12c-5.39 3.25-11.9 4.29-17.68 6.68a47.91 47.91 0 00-17.34 11.65 21.58 21.58 0 00-2.64 3.39 1.91 1.91 0 00.67 2.57 13.36 13.36 0 016 8.57 20.56 20.56 0 01.51 2.5l-7.55-3a1.89 1.89 0 00-2.38 1.81c0 1.33.09 2.65.14 4l-7.65-5.09a1.89 1.89 0 00-2.83 1.61v5.27l-.59-.37-.68-.46-.31-.23-.08-.07a7 7 0 01-2.17-2.67 5.08 5.08 0 01-.32-.71c-.15-.41-.28-.83-.4-1.25a16.31 16.31 0 01-.5-2.86 45.16 45.16 0 01.49-10.45 2 2 0 00-1.32-2.26 1.89 1.89 0 00-2.31 1.31c-.3 2.12-.6 4.37-.74 6.64-5-4.28-5.92-12.16-6.43-18.39-.1-1.34-.18-2.68-.35-4a3.92 3.92 0 00-1-2.47 36.11 36.11 0 01-2.93-4A37.75 37.75 0 0132 147.07a114.82 114.82 0 01-.32-13.63 54.4 54.4 0 0017.52 6.47c2.36.45 3.37-3.17 1-3.61-12.31-2.3-24.7-8.75-30.54-20.16h5.87c1.73 0 4.24.15 4.39-2.2a17.49 17.49 0 00-.77-4.69c-.1-.43-.19-.87-.29-1.3l2.63.81 4.93 1.59a1.87 1.87 0 002.3-1.31q.73-3.44 1.46-6.89c1.27 1.13 2.49 2.33 3.65 3.58a73.22 73.22 0 0112 18.09c1 2.17 4.28.27 3.23-1.89a76.27 76.27 0 00-12.56-18.85 40.62 40.62 0 00-6.54-6 1.9 1.9 0 00-2.76 1.13c-.55 2.61-1.11 5.23-1.66 7.84-1.68-.58-3.38-1.18-5.08-1.69-.06-1-.11-2-.12-3a32.68 32.68 0 01.13-3.57 8.21 8.21 0 01.23-1.31c.05-.18.12-.36.16-.55-.1.4.32-.41 0 0A1.89 1.89 0 0029 92.89c-1.87.3-3.77 1.87-5.29 2.89l-.36.26a16.66 16.66 0 015.74-7.87 1.9 1.9 0 00-.95-3.5c-2.92-.21-5.82-.12-8.7-.32 1.69-1 3.38-2.06 5-3.23a34.88 34.88 0 007.26-7.69c7 1.37 14.18 2.25 21.3 2.92a250 250 0 0033.62 1.06c15.67-.64 32-2.83 46.37-9.48 4.49 10.07 4.1 22.6-1.54 32.14zM134 63.24a76.5 76.5 0 01-17 6.28 139.68 139.68 0 01-24 3.77 223 223 0 01-30.91 0 233.69 233.69 0 01-29.79-3.55c-7-1.36-14.29-3-20.72-6.29a20.24 20.24 0 01-5.83-4 7.42 7.42 0 01-1.5-2.11l-.09-.17a.44.44 0 000-.08 8.176 8.176 0 01-.36-1.3v-.35a7.85 7.85 0 010-.84 50.25 50.25 0 0020-2.8 65.6 65.6 0 0011.92-6.37c.63-.39 1.26-.77 1.91-1.11l.36 3.5a13.89 13.89 0 00-3.51 1.61c-1.34.9-2.29 2.15-2 3.83a13.14 13.14 0 001.52 3.8c.42.77.87 1.52 1.32 2.27a1.93 1.93 0 00-.82 3.52A29.62 29.62 0 0042.23 66a40 40 0 0018.89.43c2.36-.51 1.36-4.13-1-3.62-.1 0-.2 0-.3.06a30.52 30.52 0 00-3.57-7.92A120 120 0 0090.73 54a68.21 68.21 0 0015.86-4.42c3.18-1.38 7.11-3.42 8.54-6.78 1.23-2.88 1.06-6.3.6-9.46a1.79 1.79 0 00-.12-.83l-.12-.72a31.44 31.44 0 003.89-1.7c.55-.27.83-.39 1.14-.51a15.12 15.12 0 011.62-.49 12.13 12.13 0 013.65-.15c3.86.36 7.62 2 11.09 3.64a36.91 36.91 0 017.77 4.62 11.89 11.89 0 014.41 6.37c.67 3-.75 6.21-2.45 8.76A33.2 33.2 0 01134 63.24zM115.22 50.16h.07-.07z"
                      />
                      <path
                        d="M48.18 38.63a1.88 1.88 0 000-3.75 1.88 1.88 0 000 3.75zM56.65 39.59a1.88 1.88 0 000-3.75 1.88 1.88 0 000 3.75zM37.3 169.61l-.17-.18c0 .06 0 .13.17.19z"
                      />
                      <path
                        class="prefix__cls-4"
                        d="M95.69 23.09c-.93-3.61-6.78-7.13-10.09-8.37a4.94 4.94 0 00-4 .26s1.54 1 3.86 2.64c2.14 1.51 5.24 3.68 6 6.55s-.88 5.72-2 8.1-2.2 4.4-2.2 4.4a4 4 0 003.17-.85c2.51-2.62 6.33-8.61 5.26-12.73z"
                      />
                      <path
                        class="prefix__cls-4"
                        d="M88.48 24.59c-.66-2.53-4.79-5-7.12-5.89a3.45 3.45 0 00-2.83.18s1.08.69 2.72 1.86 3.69 2.6 4.22 4.62-.62 4-1.43 5.7-1.55 3.11-1.55 3.11a2.85 2.85 0 002.24-.6c1.81-1.85 4.5-6.07 3.75-8.98z"
                      />
                    </g>
                  </g>
                </svg>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
